auction-server/src/kernel/pyth_lazer.rs

```diff
@@ -0,0 +1,312 @@
+                        }
+                        Message::Binary(binary) => {
+                            let response: Result<MessageType, serde_json::Error> = serde_json::from_slice(binary.as_slice());
+                            match response {
+                                Ok(response) => response,
+                                Err(err) => {
+                                    tracing::warn!(binary = ?binary, error = ?err, "Failed to parse lazer binary message");
+                                    continue;
+                                }
+                            }
+                        }
+                        Message::Close(_) => break,
+                        Message::Pong(_) => continue,
+                        Message::Ping(data) => {
+                            let _ = ws_stream.send(Message::Pong(data)).await;
+                            continue;
+                        },
+                        Message::Frame(_) => continue,
+                    };
+
+                    match message {
+                        MessageType::Subscribed(_) => continue,
+                        MessageType::StreamUpdated(update) => {
+                            if let Err(err) = update_sender.send(update) {
+                                tracing::warn!(error = ?err, "Failed to broadcast lazer update message");
+                            }
+                            continue;
+                        }
+                    }
+                }
+                _  = connection_check.tick() => {
+                    if latest_update.elapsed() > maximum_inactivity {
+                        tracing::warn!("Lazer connection inactive for too long, closing connection");
+                        break;
+                    }
+                },
+            }
+        }
+    }
+}
+
+impl PythLazer {
+    pub fn new(url: Url, api_key: String, price_feed_ids: Vec<PriceFeed>) -> Self {
+        let mut price_feeds = HashMap::new();
+        for price_feed in price_feed_ids {
+            price_feeds.insert(price_feed.id, price_feed);
+        }
+        PythLazer {
+            url,
+            api_key,
+            price_feeds,
+        }
+    }
+
+    pub async fn subscribe(&self) -> anyhow::Result<broadcast::Receiver<StreamUpdatedMessage>> {
+        let mut request = self.url.as_str().into_client_request()?;
+        let bearer_token = format!("Bearer {}", self.api_key);
+        request
+            .headers_mut()
+            .insert("Authorization", bearer_token.parse()?);
+
+        let (mut ws_stream, _) = connect_async(request).await?;
+        let message = serde_json::to_string(&SubscribeRequest {
+            id:             1,
+            request_type:   SubscribeRequestType::Subscribe,
+            price_feed_ids: self.price_feeds.keys().cloned().collect(),
+            properties:     vec![SubscribeRequestProperties::Price],
+            channel:        SubscribeRequestChannel::FixedRate200Ms,
```

Comment:
why always 200 ms? you can pull the lazer feeds' metadata and get the highest frequency channel if you want?